.github/workflows/ios-regression.yml

```diff
@@ -0,0 +1,215 @@
+      IOS_EIGHTH_SIMULATOR: '1C61A0B6-3085-46B6-9B2E-6E9A1B594944'
+      IOS_NINTH_SIMULATOR: '7C69D129-BD40-4BC0-9A0D-6A7E990F23E8'
+      IOS_TENTH_SIMULATOR: '275E0C5B-2458-4490-831D-4CE86C8EC704'
+      IOS_ELEVENTH_SIMULATOR: 'EA5C5BCB-1E3E-4FFC-88BB-8D1DE354BC3B'
+      IOS_TWELFTH_SIMULATOR: '41514F87-DC49-4C81-B60C-03D1F8653A42'
+      # echo "SHARD_NUMBER ${{ github.event.inputs.SHARD_NUMBER }}"
+      # echo "SHARD_COUNT ${{ github.event.inputs.SHARD_COUNT }}"
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Runner Details
+        run: |
+          echo "BRANCH_TO_CHECKOUT ${{ github.event.inputs.BRANCH_TO_CHECKOUT }}"
+          echo "APPIUM_REPO ${{ github.event.inputs.APPIUM_REPO }}"
+          echo "APK_URL ${{ github.event.inputs.APK_URL }}"
+          echo "RISK ${{ github.event.inputs.RISK }}"
+
+      - name: Download ipa
+        run: |
+          curl -o session-ios.sim.tar.xz ${{ github.event.inputs.APK_URL }}
+          ls
+          pwd
+
+      - name: Inspect archive contents
+        run: |
+          tar -tf session-ios.sim.tar.xz
+
+      - name: Extract app
+        run: |
+          tar xf session-ios.sim.tar.xz
+          mkdir -p extracted
+          mv *.app extracted/ || echo "No .app files found to move"
+          ls extracted
+
+      - name: Verify extracted files
+        run: |
+          if [ ! -d "extracted" ]; then
+            echo "Extraction failed: extracted directory not found"
+            exit 1
+          fi
+          ls extracted
+
+      - name: Rename extracted folder
+        run: |
+          mkdir -p extracted
+          mv session-ios extracted || echo "No session-ios directory to move"
+          ls extracted
+
+      - uses: actions/checkout@v4
+        with:
+          repository: ${{ github.event.inputs.APPIUM_REPO }}
+          ref: ${{ github.event.inputs.BRANCH_TO_CHECKOUT }}
+          path: 'forked-session-appium'
+
+      - uses: actions/setup-node@v4
+        with:
+          # node-version-file: 'forked-session-appium/.nvmrc'
+          node-version: 18.15.0
+
+      - name: Install yarn
+        run: |
+          npm install -g yarn
+          corepack enable
+          yarn set version 4.1.1
+      - name: Install test dependencies
+        run: |
+          cd forked-session-appium
+          ls
+          git status
+          touch yarn.lock
+          yarn install --immutable
+      - name: Start Appium server
+        run: |
+          cd forked-session-appium
+          yarn "./node_modules/.bin/appium server --use-drivers=uiautomator2,xcuitest --port 8110 --allow-cors"
+      - name: Build the iOS tests
+        run: |
+          cd forked-session-appium
+          yarn tsc
+      - name: Start simulators
+        shell: bash
+        run: |
+          source ./scripts/ci.sh
+          start_simulators_from_env_iOS
+      - name: List all tests
+        run: |
+          cd forked-session-appium
+          pwd
+          npx playwright test --list
```

Comment:
maybe we should add the filter on the risk selected here, just so we dont' get confused with the count of tests to run/list of them

Reply:
same comment for android btw